profiles/tests/test_gql_profiles_query.py

```diff
@@ -141,14 +141,27 @@ def test_staff_user_can_filter_profiles_by_first_name(user_gql_client, group, se
             profiles(firstName: $firstName) {
                 count
                 totalCount
+                edges {
+                    node {
+                        firstName
+                    }
+                }
             }
         }
     """
 
-    expected_data = {"profiles": {"count": 1, "totalCount": 2}}
+    expected_data = {
+        "profiles": {
+            "count": 1,
+            "totalCount": 2,
+            "edges": [{"node": {"firstName": profile_2.first_name}}],
+        }
+    }
 
     executed = user_gql_client.execute(
-        query, variables={"firstName": profile_2.first_name}, service=service,
+        query,
+        variables={"firstName": profile_2.first_name[1:-1].upper()},
```

Comment:
The test data generation isn't completely random. There's an [autouse fixture that initializes the random generator](https://github.com/City-of-Helsinki/open-city-profile/blob/develop/open_city_profile/tests/conftest.py#L69-L71) with a fixed value before every test. Hence the tests always get the same generated values.

Nevertheless, your point is still valid. What if I just cut the first character away? It would even simplify the code a bit and it still more or less achieves the same thing.

Reply:
Changed the search string generation to only remove the first character.

Reply:
Yeah, that should work.